src/sentry/models/avatars/base.py

```diff
@@ -35,17 +38,31 @@ def save(self, *args, **kwargs):
         return super().save(*args, **kwargs)
 
     def get_file(self):
-        from sentry.models import File
-
-        if self.file_id is None:
-            return None
+        # Favor control_file_id if it exists and is set.
+        # Otherwise fallback to file_id. If still None, return.
+        file_class = self.file_class()
+        file_id = getattr(self, self.file_fk())
+        if file_id is None:
+            file_id = self.file_id
+            file_class = File
+            if file_id is None:
+                return None
+            copy_file_to_control_and_update_model.apply_async(
```

Comment:
Should we have an option/feature flag to control the flow of work into this queue? Without that upon deploy, we'll get a thundering herd of file move operations, and there will be lots of dupes in those jobs as frequently used avatars have jobs spawned while previous jobs are still in the queue.